test/modules/mongodb_cluster.go

```diff
@@ -0,0 +1,126 @@
+package modules
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+	"testing"
+
+	aws_sdk "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/gruntwork-io/terratest/modules/aws"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	test_structure "github.com/gruntwork-io/terratest/modules/test-structure"
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/atlas-sdk/v20231001002/admin"
+)
+
+func DeployMongoDBCluster(t *testing.T, workingDir string) {
+	// Get the role arn from env
+	roleArn := os.Getenv("TF_VAR_mongodb_role_arn")
+	// Generate a random id
+	randomID := random.UniqueId()
+	terraformOptions := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+		TerraformDir: workingDir,
+		Vars: map[string]interface{}{
+			"mongodb_role_arn": roleArn,
+			"random_id":        randomID,
+		},
+	})
+
+	// Save the options so later test stages can use them
+	test_structure.SaveTerraformOptions(t, workingDir, terraformOptions)
+}
+
+func ValidateMongoDBCluster(t *testing.T, workingDir string) {
+	terraformOptions := test_structure.LoadTerraformOptions(t, workingDir)
+	// Get public key and private key from secrets manager
+	secretArn := os.Getenv("MONGODB_SECRET_ARN")
+	secretString := aws.GetSecretValue(t, "us-east-1", secretArn)
+	var keys struct {
+		PrivateKey string `json:"private_key"`
+		PublicKey  string `json:"public_key"`
+	}
+
+	err := json.Unmarshal([]byte(secretString), &keys)
+	if err != nil {
+		t.Fatalf("Error when unmarshalling json: %v", err)
+	}
+
+	apiKey := keys.PublicKey
+	apiSecret := keys.PrivateKey
+	// Create admin client
+	sdk, err := admin.NewClient(admin.UseDigestAuth(apiKey, apiSecret))
+	if err != nil {
+		log.Fatalf("Error when instantiating new client: %v", err)
+	}
+
+	// CASE: MongoDB Cluster outputs are correct
+	// Get the mongodb base uri
+	mongoURIs := terraform.Output(t, terraformOptions, "cluster_mongodb_base_uri")
```

Comment:
This should only be a string not a list, if it is in fact a list then we should rename the output name.

Reply:
Its a comma separated string